### ISSUE TITLE
"New connection - Action" Segment event should feature the name of the connectors used, not the user's name of the source an destination

### ISSUE THREAD

**Issue by johnlafleur** [airbytehq/airbyte#2087]
Specs:  https://docs.google.com/spreadsheets/d/1qQPMBdz86yMBiyXpjw91D3HoP6hN7yIPRmLrF7xWOgA/edit#gid=1567609168, 

Unfortunately, "connector_destination_definition" seems to be the name given by the user for this definition. But in "connector_destination" (which is missing, so I'm guessing the _definition replaced it), we want to know the name of the tool used. 

You can see that here: https://analytics.amplitude.com/daxtarity/chart/7i04ang/edit/zspyz7v
or in the screenshot below
<img width="738" alt="Screen Shot 2021-02-17 at 9 08 32 AM" src="https://user-images.githubusercontent.com/68561602/108127798-e3750300-70ff-11eb-940d-096847e7a6be.png">

@cgardens

**Comment by cgardens** [comment]
@Jamakase looking into it, I think the confusion here is on lines like this:
[line number](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx#L90): `connector_source_definition: source?.sourceName,` => `connector_source_definition: source?.sourceDefinitionName,`

Here are the instances i could find. In all of these cases we want to show the definition name instead of name of the connector itself.
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx#L90
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/pages/SourcesPage/pages/CreateSourcePage/components/SourceForm.tsx#L59
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/hooks/services/useConnectionHook.tsx#L133
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx#L92
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/hooks/services/useConnectionHook.tsx#L135
* https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/pages/DestinationPage/pages/CreateDestinationPage/components/DestinationForm.tsx#L59

in other words, wherever we are sending to segment a `connector_source_definition` or a `connector_destination_definition`, we should be sending the definition name.

i suspect that this might be a problem for you on the frontend and that you'll need more information from the API. e.g. 
* add destination definition name to `DestinationRead`
* add source definition name to `SourceRead`
are those sufficient? I wasn't quite sure in reading `CreateConnectionContent` is this case could be handled by adding those fields to the previous 2 objects i mentioned.

**Comment by Jamakase** [comment]
@cgardens Not quite sure what's wrong here - we already send the definition name, not the destination/source name.

fields of source are next

`source.name` is Source name defined by user. And `source.sourceName` is name of definition ( quite confusing ) but that's what we receive from API :) 

![image](https://user-images.githubusercontent.com/3767150/109039615-91377180-76ff-11eb-8325-3475d33e9d9d.png)

**Comment by cgardens** [comment]
@johnlafleur This should have been fixed ~14 days ago ([commit](https://github.com/airbytehq/airbyte/commit/a8ad18455124188e94433b130801a5d5c02c57ab)). Is it possible that the data in amplitude is old? . I'm inclined to say that this can be closed unless you can find evidence that new versions of airbyte are sending the wrong names. Based on the code it seems that should not be the case.

**Comment by johnlafleur** [comment]
I'm sorry this is not what I see on Amplitude in the last 7 days. See https://analytics.amplitude.com/daxtarity/chart/44a68f9
I also can't explain how we can have connections created without any connector_source_definition...

<img width="1350" alt="Screen Shot 2021-02-25 at 10 23 21 AM" src="https://user-images.githubusercontent.com/68561602/109079749-7f7fb980-7753-11eb-9f8b-a74fed969786.png">

**Comment by cgardens** [comment]
I'm looking in amplitude and the ones that use the Postgres name (the name we would expect are all using a new version):
![Screen Shot 2021-02-24 at 4 20 33 PM](https://user-images.githubusercontent.com/9092207/109084421-9d4e0f80-76bc-11eb-9eca-2d8b30fd57ff.png)

When I look at events with whacky names i see that they seem to be old versions.
![Screen Shot 2021-02-24 at 4 19 49 PM](https://user-images.githubusercontent.com/9092207/109084453-ab9c2b80-76bc-11eb-8f6a-87cb8c86b6ab.png)

Until people are not using old versions anymore we are still going to get events with the wrong names. Not too much we can do about this except encourage people to upgrade. Based on the evidence thus far I think this problem is fixed for new versions.
